sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ExchangeCoordinator.scala

```diff
@@ -83,16 +83,17 @@ import org.apache.spark.sql.execution.{ShuffledRowRDD, SparkPlan}
  *  - post-shuffle partition 3: pre-shuffle partition 3 and 4 (size 50 MB)
  */
 class ExchangeCoordinator(
-    numExchanges: Int,
     advisoryTargetPostShuffleInputSize: Long,
     minNumPostShufflePartitions: Option[Int] = None)
   extends Logging {
 
   // The registered Exchange operators.
   private[this] val exchanges = ArrayBuffer[ShuffleExchangeExec]()
```

Comment:
we should remove the exchange being reused.

Reply:
In the current fix, this `exchanges` doesn't already have reused exchanges, e.g., in the example case described above, `exchanges.size` already has been `1`.

Reply:
ah you are right